packages/server/src/api/controllers/table.js

```diff
@@ -41,6 +41,18 @@ exports.save = async function(ctx) {
     oldTable = await db.get(ctx.request.body._id)
   }
 
+  // make sure that types don't change of a column, have to remove
+  // the column if you want to change the type
+  if (oldTable && oldTable.schema) {
+    for (let propKey of Object.keys(tableToSave.schema)) {
```

Comment:
NAB: `for let propKey in tableToSave.schema`

Reply:
Generally I prefer the `for ... of Object.keys` approach as there is no risk of iterating prototype properties that were not intended. Why these should be plain objects if we were to ever use a more OO design for structures like tables then it would be very likely functions and so on from base classes would show up in a `for ... in` loop, if prototype information is required it is better to be explicit in requesting it.

Also Webstorm gets really angry when `for ... in` loops are used without performing a proper `Object.hasOwnProperty` check to make sure that the properties you're iterating actually own the properties.

Reply:
Yep, makes sense in an OO context if we were to move to that 👍